api/src/pcapi/core/offerers/repository.py

```diff
@@ -392,3 +395,22 @@ def find_available_reimbursement_points_for_offerer(offerer_id: int) -> list[mod
 
 def get_venue_by_id(venue_id: int) -> models.Venue:
     return models.Venue.query.get(venue_id)
+
+
+def find_offerers_validated_3_days_ago_with_no_venues() -> list[models.Offerer]:
+    three_days_ago = datetime.utcnow() - timedelta(days=3)
+    three_days_ago_min = datetime.combine(three_days_ago, time.min)
+    three_days_ago_max = datetime.combine(three_days_ago, time.max)
+
+    subquery = db.session.query(models.Venue.managingOffererId)
+
+    return (
+        db.session.query(models.Offerer)
+        .filter(models.Offerer.id.not_in(subquery))
+        .filter(
+            models.Offerer.isActive.is_(True),
+            models.Offerer.dateValidated >= three_days_ago_min,
+            models.Offerer.dateValidated <= three_days_ago_max,
+        )
+        .all()
```

Comment:
Si on n'a pas impérativement besoin d'avoir une liste, est-ce qu'on éviter les `.all()` ? Mais c'est une discussion peut-être à avoir plus globalement avec le reste de l'équipe.

Reply:
Pas compris la remarque, on peut avoir une ou plusieurs structures validées il y a 3 jours sans lieu

Reply:
Si tu ne fais qu'itérer sur l'objet obtenu, l'appel à `.all()` ne change rien : dans un cas tu auras une liste, dans l'autre un objet `Query` (`BaseQuery` ?) sur lequel tu peux itérer.

Petit avantages : tu peux facilement récupérer cet objet pour ajouter des filtres plus tard ou des options comme `.yield_per` ou encore afficher le SQL généré.

Après c'est du détail. Mais personnellement, je trouve que si on n'a pas impérativement besoin d'une liste, l'appel à `.all()` n'apporte rien (pas grand chose ?) et à quelques petits inconvénients.

Comme tu préfères :)